modal/sandbox.py

```diff
@@ -244,6 +257,24 @@ async def wait(self, raise_on_termination: bool = True):
                     raise SandboxTerminatedError()
                 break
 
+    async def tunnels(self) -> List[api_pb2.TunnelData]:
+        """Get the tunnels for the sandbox."""
+
+        if self._tunnels:
+            return self._tunnels
+
+        while True:
+            req = api_pb2.SandboxGetTunnelsRequest(sandbox_id=self.object_id, timeout=50)
+            resp = await retry_transient_errors(self._client.stub.SandboxGetTunnels, req)
+
+            # If we couldn't get the tunnels in time, try again.
+            if resp.result.status == api_pb2.GenericResult.GENERIC_STATUS_TIMEOUT:
+                continue
```

Comment:
Shoudl we give up at some point to avoid endlessly hanging?

Reply:
I'm just going to not retry. If the tunnels don't come up inside 50 seconds, it's probably an error. The user can override timeout if they care.